src/pet.js

```diff
@@ -0,0 +1,70 @@
+const MAXIMUM_FITNESS = 10;
+
+function Pet(name) {
+  this.name = name;
+  this.age = 0;
+  this.hunger = 0;
+  this.fitness = 10;
+  this.children = [];
+}
+
+Pet.prototype = {
+  get isAlive() {
+    if (this.fitness <= 0) return false;
+    else if (this.hunger >= 10) return false;
+    else if (this.age >= 30) return false;
+    else return true;
+  },
+};
+
+Pet.prototype.growUp = function () {
+  if (!this.isAlive) throw new Error("Your pet is no longer alive :(");
+
+  this.age += 1;
+  this.hunger += 5;
+  this.fitness -= 3;
+};
+
+Pet.prototype.walk = function () {
+  if (!this.isAlive) throw new Error("Your pet is no longer alive :(");
+
+  const newFitness = this.fitness + 4;
+  if (newFitness > MAXIMUM_FITNESS) {
+    this.fitness = MAXIMUM_FITNESS;
+  } else {
+    this.fitness = newFitness;
+  }
```

Comment:
This could be refactored a bit. In essence you are adding a value to the existing fitness, without allowing it go over a set limit. You can do this with Math.min, which will return the lowest of the values passed to it:
```
this.fitness = Math.min(this.fitness + 4, MAXIMUM_FITNESS)
```